src/jit/lsra.cpp

```diff
@@ -5679,11 +5679,12 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
 
     if (availablePhysRegInterval != nullptr)
     {
-        intervalToUnassign = availablePhysRegInterval->assignedInterval;
         if (isAssigned(availablePhysRegInterval ARM_ARG(currentInterval->registerType)))
         {
             unassignPhysReg(availablePhysRegInterval ARM_ARG(currentInterval->registerType));
-            if (bestScore & VALUE_AVAILABLE)
+            intervalToUnassign = availablePhysRegInterval->assignedInterval;
```

Comment:
@CarolEidt OK. I have changed the position to before calling unassignPhysReg like what you said. Thank you.!